lib/IRGen/GenStruct.cpp

```diff
@@ -61,6 +63,31 @@ static StructTypeInfoKind getStructTypeInfoKind(const TypeInfo &type) {
   return (StructTypeInfoKind) type.getSubclassKind();
 }
 
+clang::CXXDestructorDecl *swift::irgen::getCXXDestructor(SILType type) {
+  clang::CXXDestructorDecl *destructor = nullptr;
+  if (!type.getStructOrBoundGenericStruct() ||
```

Comment:
Suggest storing the result of `getStructOrBoundGenericStruct` in a variable to avoid duplicate calls.